internals/daemon/daemon_test.go

```diff
@@ -1305,6 +1313,50 @@ func (s *daemonSuite) TestHTTPAPI(c *C) {
 	c.Assert(err, ErrorMatches, ".* connection refused")
 }
 
+func (s *daemonSuite) TestHTTPSAPI(c *C) {
+	s.httpsAddress = ":0" // Go will choose port (use listener.Addr() to find it)
+	d := s.newDaemon(c)
+	d.Init()
+	c.Assert(d.Start(), IsNil)
+	port := d.httpsListener.Addr().(*net.TCPAddr).Port
+
+	httpsClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	request, err := http.NewRequest("GET", fmt.Sprintf("https://localhost:%d/v1/health", port), nil)
+	c.Assert(err, IsNil)
+	response, err := httpsClient.Do(request)
+	c.Assert(err, IsNil)
+	c.Assert(response.StatusCode, Equals, http.StatusOK)
+	var m map[string]any
+	err = json.NewDecoder(response.Body).Decode(&m)
+	c.Assert(err, IsNil)
+	c.Assert(m, DeepEquals, map[string]any{
+		"type":        "sync",
+		"status-code": float64(http.StatusOK),
+		"status":      "OK",
+		"result": map[string]any{
+			"healthy": true,
+		},
+	})
+
+	request, err = http.NewRequest("GET", fmt.Sprintf("https://localhost:%d/v1/checks", port), nil)
+	c.Assert(err, IsNil)
+	response, err = httpsClient.Do(request)
+	c.Assert(err, IsNil)
+	c.Assert(response.StatusCode, Equals, http.StatusUnauthorized)
+
+	err = d.Stop(nil)
```

Comment:
I see that `TestHTTPAPI` does this the same way, but should this be done in a `defer`, as in `defer d.Stop(nil)` just after the `d.Start()` call?

Reply:
Note that there is one test case deliberately done still after the daemon is stopped, so I think that piece has to stay.

However, I've added a conditional defer to handle the case where the test fail early, e.g. one of the ```c.Assert``` cases fail, resulting in a ```runtime.Goexit``` which triggers the defer chain.

I think this should make it reliable, but do let me know if what I have done makes sense for you?

Reply:
Resolving.